pre_commit/clientlib.py

```diff
@@ -112,6 +112,24 @@ def validate_manifest_main(argv: Optional[Sequence[str]] = None) -> int:
 META = 'meta'
 
 
+class CheckMutableRev:
+    key = 'rev'
+
+    def check(self, dct: Dict[str, Any]) -> None:
+        if any(
+            mutable in dct.get(self.key, '')
+            for mutable in {'HEAD', 'stable', 'master'}
+        ):
+            logger.warning(
+                f"The {self.key!r} field of repo {dct.get('repo')!r} "
```

Comment:
repo is required by this point

Reply:
changed it to `dct['repo']` if that's what you referred to here